src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -40,12 +41,52 @@ namespace Microsoft.VisualStudio.LanguageServices.Setup
     [ProvideMenuResource("Menus.ctmenu", version: 16)]
     internal class RoslynPackage : AbstractPackage
     {
+        // note: must be <= 31 characters long
+        private const string DecompilerEulaOptionKey = "ILSpy-234190A6EE66";
```

Comment:
What's with the magic number?

Reply:
:arrow_right: I rebased this change to fix merge conflicts, and the only change of substance was the addition of this comment for the constant:

> The randomly-generated key name is used for serializing the ILSpy decompiler EULA preference to the .SUO file. It doesn't have any semantic meaning, but is intended to not conflict with any other extension that might be saving an "ILSpy" named stream to the same file.